### PR TITLE
[IMP] mail: introduce AutocompleteInputView/customClass

### PR DESCRIPTION
**Pull request by alexkuhn**
Task-2817076

### CODE REVIEW

File: addons/mail/static/src/components/messaging_menu/messaging_menu.js [hunk 1]
```diff
@@ -1,26 +1,11 @@
...
 const { Component } = owl;
 
 export class MessagingMenu extends Component {
 
-    /**
-     * @override
-     */
-    setup() {
-        super.setup();
-        useComponentToModel({ fieldName: 'component', modelName: 'MessagingMenu' });
```
Comment (seb-odoo) [inline addons/mail/static/src/components/messaging_menu/messaging_menu.js:15]: unrelated diff?

File: addons/mail/static/src/models/messaging_menu.js [hunk 2]
```diff
@@ -231,5 +231,13 @@ registerModel({
             inverse: 'messagingMenuAsPinnedAndUnreadChannel',
             readonly: true,
         }),
+        /**
+         * global JS generated ID for this record view. Useful to provide a
+         * custom class to autocomplete input, so that click in an autocomplete
+         * item is not considered as a click away from messaging menu in mobile.
+         */
+        viewId: attr({
+            default: _.uniqueId('o_messagingMenu_'),
```
Comment (seb-odoo) [inline addons/mail/static/src/models/messaging_menu.js:240]: must be a compute, default is static

maybe just move in compute of custom class?

File: addons/mail/static/src/models/autocomplete_input_view.js [hunk 3]
```diff
@@ -22,6 +22,23 @@ registerModel({
...
+        /**
+         * @private
+         * @returns {string|FieldCommand}
+         */
+        _computeCustomClass() {
+            if (this.discussSidebarCategoryOwnerAsAddingItem) {
+                if (this.discussSidebarCategoryOwnerAsAddingItem === this.messaging.discuss.categoryChannel) {
+                    return 'o_DiscussSidebarCategory_newChannelAutocompleteSuggestions';
+                } else {
+                    return false;
```
Comment (seb-odoo) [inline addons/mail/static/src/models/autocomplete_input_view.js:34]: clear instead?

**Review by seb-odoo — CHANGES_REQUESTED** [review]

File: addons/mail/static/src/components/messaging_menu/messaging_menu.js (hunk 1 above)
Comment (alexkuhn) [reply addons/mail/static/src/components/messaging_menu/messaging_menu.js:15]: Changes done

File: addons/mail/static/src/models/autocomplete_input_view.js (hunk 3 above)
Comment (alexkuhn) [reply addons/mail/static/src/models/autocomplete_input_view.js:34]: Changes done.

File: addons/mail/static/src/models/messaging_menu.js (hunk 2 above)
Comment (alexkuhn) [reply addons/mail/static/src/models/messaging_menu.js]: Changes done

File: addons/mail/static/src/models/autocomplete_input_view.js [hunk 4]
```diff
@@ -51,6 +51,23 @@ registerModel({
...
+         * @private
+         * @returns {string|FieldCommand}
+         */
+        _computeCustomClass() {
+            if (this.discussSidebarCategoryOwnerAsAddingItem) {
+                if (this.discussSidebarCategoryOwnerAsAddingItem === this.messaging.discuss.categoryChannel) {
+                    return 'o_DiscussSidebarCategory_newChannelAutocompleteSuggestions';
+                } else {
+                    return clear();
+                }
```
Comment (seb-odoo) [inline addons/mail/static/src/models/autocomplete_input_view.js:64]: This clear is actually useless in regard to the one at the end of the method

**Review by seb-odoo — APPROVED** [review]
Ok except minor comment above

File: addons/mail/static/src/models/autocomplete_input_view.js (hunk 4 above)
Comment (alexkuhn) [reply addons/mail/static/src/models/autocomplete_input_view.js]: Change done

**Comment by alexkuhn** [comment]
Moved to grouped PR: https://github.com/odoo/odoo/pull/90635
